db.py

```diff
@@ -1,53 +1,60 @@
 """A medium-faithful port of https://github.com/weinberg/SQLToy to Python"""
 
+from typing import Any, Callable, Dict, Iterable, Optional, Set, Tuple
+
 
```

Comment:
Let's do `Row = Dict[str, Any]` and `RowPredicate = Callable[[Row], bool]` to make types easier to read

Reply:
Adding new type aliases so we can use those throughout the new type annotations. This will make nearly every annotation much shorter